### PR TITLE
[14.0] [mig]pos_fixed_discount

### PR DESCRIPTION
**Pull request by hkapatel-initos**

### CODE REVIEW

File: pos_fixed_discount/static/src/js/discount.js [hunk 1]
```diff
@@ -1,33 +1,38 @@
-odoo.define("pos_fixed_discount.pos_fixed_discount", function (require) {
+odoo.define("pos_fixed_discount.FixedDiscountButton", function (require) {
```
Comment (ivantodorovich) [inline pos_fixed_discount/static/src/js/discount.js]: Please rename this `js` file to `FixedDiscountButton.js` :)

File: pos_fixed_discount/static/src/js/discount.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        }
+        async onClick() {
+            var self = this;
+            const {confirmed, payload} = await this.showPopup("NumberPopup", {
+                title: this.env._t("Discount Amount"),
+                startingValue: 0,
+            });
+            if (confirmed) {
+                var val = Math.round(Math.max(0, Math.min(100, parseFloat(payload))));
+                self.apply_discount(val);
```
Comment (ivantodorovich) [inline pos_fixed_discount/static/src/js/discount.js]:
```suggestion
                this.apply_discount(val);
```

You can do this and drop the `var self = this;` line at the top

File: pos_fixed_discount/static/src/js/discount.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+            // Remove existing discounts
+            var i = 0;
+            while (i < lines.length) {
+                if (lines[i].get_product() === product) {
+                    order.remove_orderline(lines[i]);
+                } else {
+                    i++;
+                }
+            }
```
Comment (ivantodorovich) [inline pos_fixed_discount/static/src/js/discount.js]:
```suggestion
            // Remove existing discounts
            for (const line of lines) {
                if (line.get_product() === product) {
                    order.remove_orderline(line);
                }
            }
```

Doesn't this do the same?

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

**Review by fshah-initos — COMMENTED** [review]
LGTM

**Review by fshah-initos — APPROVED** [review]

**Review by ivantodorovich — APPROVED** [review]

**Comment by ivantodorovich** [comment]
I don't know what you pushed, but the bot was already merging :sweat_smile: I'm not sure what's gonna happen now, maybe it'll see the change and cancel

If not, create a new PR after this one's merged pls

**Pull request closed without merging**